core/core.go

```diff
@@ -0,0 +1,132 @@
+package core
+
+import (
+	"fmt"
+
+	"github.com/DE-labtory/cleisthenes"
+
+	"github.com/DE-labtory/cleisthenes/config"
+	"github.com/DE-labtory/cleisthenes/honeybadger"
+)
+
+type handler struct {
+	handleFunc func(msg cleisthenes.Message)
+}
+
+func newHandler(handleFunc func(cleisthenes.Message)) *handler {
+	return &handler{
+		handleFunc: handleFunc,
+	}
+}
+
+func (h *handler) ServeRequest(msg cleisthenes.Message) {
+	h.handleFunc(msg)
+}
+
+type Hbbft interface {
+	Propose(tx cleisthenes.Transaction) error
+	Run()
+}
+
+type Node struct {
+	addr            cleisthenes.Address
+	txQueueManager  cleisthenes.TxQueueManager
+	batchReceiver   cleisthenes.BatchReceiver
+	messageEndpoint cleisthenes.MessageEndpoint
+	server          *cleisthenes.GrpcServer
+	client          *cleisthenes.GrpcClient
+	connPool        *cleisthenes.ConnectionPool
+	memberMap       *cleisthenes.MemberMap
+}
+
+func New() (Hbbft, error) {
+	conf := config.Get()
+
+	addr, err := cleisthenes.ToAddress(conf.Identity.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	txQueue := cleisthenes.NewTxQueue()
+	hb := honeybadger.New()
+
+	return &Node{
+		addr:            addr,
+		txQueueManager:  cleisthenes.NewDefaultTxQueueManager(txQueue, hb),
+		batchReceiver:   cleisthenes.NewBatchChannel(),
+		messageEndpoint: hb,
+		server:          cleisthenes.NewServer(addr),
+		client:          cleisthenes.NewClient(),
+		connPool:        cleisthenes.NewConnectionPool(),
+		memberMap:       cleisthenes.NewMemberMap(),
+	}, nil
+}
+
+func (n *Node) Run() {
+	handler := newHandler(func(msg cleisthenes.Message) {
+		addr, err := cleisthenes.ToAddress(msg.Sender)
+		if err != nil {
+			fmt.Println(fmt.Errorf("[handler] failed to parse sender address: addr=%s", addr))
+		}
+		if err := n.messageEndpoint.HandleMessage(msg.Message); err != nil {
+			fmt.Println(fmt.Errorf("[handler] handle message failed with err: %s", err.Error()))
+		}
+	})
+
+	n.server.OnConn(func(conn cleisthenes.Connection) {
+		conn.Handle(handler)
+		if err := conn.Start(); err != nil {
+			conn.Close()
+		}
+	})
+
+	go n.server.Listen()
+}
+
+func (n *Node) Propose(tx cleisthenes.Transaction) error {
```

Comment:
`Propose` act on `HoneyBadger` not on`Node`. it send(put) tx to queue.  maybe get have better name

Reply:
`Submit` :)

Reply:
fixed!